specs/DeliverInputMessagesToHostMessageQueue.md

```diff
@@ -0,0 +1,101 @@
+# Background
+WebView2 .NET developers often encounter issues where certain default events are not raised in the WebView2 .NET control. 
+
+This issue occurs because user inputs are directly delivered to the browser, and the host application does not receive the corresponding messages. 
+
+This causes some default event handling API of .NET control not to work, such as PreProcessMessage and ProcessCmdKey in WinForms.
+
+It also prevents some keys from reaching Key Event realated APIs such as OnKeyDown and ToolStripMenuItem.
+
+In order to solve this type of issue, we provide an api which allows user inputs pass through the browser to the host app.
```

Comment:
Curious how this works. Are the input messages coming to the browser process, and then it's doing a `SendInput`/`PostInput` to the host process? Do APIs in the host process such as Win32 `GetKeyState` work?

Reply:
It was done by changing browser's window style to make it hit-transparent, and make input directly to the host process.
I forgot to mention that this API will make some APIs async.  It includes AcceleratorKeyPressed event (similar to visual hosting), and OnKeyDown event in WinForms. ::GetKeyState does not get the accurate state of keyboard in these two APIs. For other APIs, ::GetKeyState should work properly.

Reply:
Changing to the hit-testing affects mouse input, but does setting this property also affect keyboard input?

Reply:
Yes. This makes keyboard focus to be on the host side.